File: webservices/src/main/java/fieg/modulos/cobrancacliente/enums/SituacaoCobrancaCliente.java
package fieg.modulos.cobrancacliente.enums;

import fieg.core.interfaces.EnumBanco;

public enum SituacaoCobrancaCliente implements EnumBanco {

    PROVISORIO("Provisorio"),
    DEPOSITO("Deposito"),
    ISENTO("Isento"),
    PAGO_CADIN("Pago Cadin"),
    ADM_CADIN("Administrado Cadin"),
    DELETADO("Deletado"),
    AGRUPADO("Agrupado"),
    EM_ABERTO("Em Aberto"),
    PAGO_MANUAL("Pago Manual"),
    PAGO_RETORNO_BANCO("Pago Retorno Banco"),
    PAGO_PIX("Pago Pix"),
    NAO_ADM_CR5("Nao administrado CR5"),
    ESTORNADO("Estornado");

    private final String valorBanco;

    SituacaoCobrancaCliente(String valorBanco) {
        this.valorBanco = valorBanco;
    }

    public static SituacaoCobrancaCliente fromString(String situacao) {
        if ("Administrado Cadin".equalsIgnoreCase(situacao)) {
            return ADM_CADIN;
        }
        for (SituacaoCobrancaCliente s : SituacaoCobrancaCliente.values()) {
            if (s.getValorBanco().equalsIgnoreCase(situacao.trim())) {
                return s;
            }
        }
        throw new IllegalArgumentException("Situação inválida: " + situacao);
    }

    @Override
    public String getValorBanco() {
        return valorBanco;
    }
}

File: web/src/models/DTOs/Cobrancas/SituacaoCobrancaCliente.ts
export interface SituacaoCobrancaClienteDTO {
  descricao: string;
  nomeEnum: string;
}

export interface SituacaoCobrancaClienteType {
  PROVISORIO: SituacaoCobrancaClienteDTO;
  DEPOSITO: SituacaoCobrancaClienteDTO;
  ISENTO: SituacaoCobrancaClienteDTO;
  PAGO_CADIN: SituacaoCobrancaClienteDTO;
  ADM_CADIN: SituacaoCobrancaClienteDTO;
  DELETADO: SituacaoCobrancaClienteDTO;
  AGRUPADO: SituacaoCobrancaClienteDTO;
  EM_ABERTO: SituacaoCobrancaClienteDTO;
  PAGO_MANUAL: SituacaoCobrancaClienteDTO;
  PAGO_RETORNO_BANCO: SituacaoCobrancaClienteDTO;
  PAGO_PIX: SituacaoCobrancaClienteDTO;
  NAO_ADM_CR5: SituacaoCobrancaClienteDTO;
  ESTORNADO: SituacaoCobrancaClienteDTO;
  EMABERTO: SituacaoCobrancaClienteDTO;
}

export type SituacaoCobrancaCliente = keyof SituacaoCobrancaClienteType;

/**
 * Aqui, label é um nome amigável e value é o valor no Enum SituacaoCobrancaCliente no cr5-webservices-v2.
 * Constante feita para ser usada como opção de SelectControl
 */
export const situacoesCobrancaClienteValues: SituacaoCobrancaClienteType = {
  PROVISORIO: {
    nomeEnum: "PROVISORIO",
    descricao: "Provisorio",
  },
  DEPOSITO: {
    nomeEnum: "DEPOSITO",
    descricao: "Deposito",
  },
  ISENTO: {
    nomeEnum: "ISENTO",
    descricao: "Isento",
  },
  PAGO_CADIN: {
    nomeEnum: "PAGO_CADIN",
    descricao: "Pago Cadin",
  },
  ADM_CADIN: {
    nomeEnum: "ADM_CADIN",
    descricao: "Administrado Cadin",
  },
  DELETADO: {
    nomeEnum: "DELETADO",
    descricao: "Deletado",
  },
  AGRUPADO: {
    nomeEnum: "AGRUPADO",
    descricao: "Agrupado",
  },
  EM_ABERTO: {
    nomeEnum: "EM_ABERTO",
    descricao: "Em Aberto",
  },
  PAGO_MANUAL: {
    nomeEnum: "PAGO_MANUAL",
    descricao: "Pago Manual",
  },
  PAGO_RETORNO_BANCO: {
    nomeEnum: "PAGO_RETORNO_BANCO",
    descricao: "Pago Retorno Banco",
  },
  PAGO_PIX: {
    nomeEnum: "PAGO_PIX",
    descricao: "Pago Pix",
  },
  NAO_ADM_CR5: {
    nomeEnum: "NAO_ADM_CR5",
    descricao: "Nao administrado CR5",
  },
  ESTORNADO: {
    nomeEnum: "ESTORNADO",
    descricao: "Estornado",
  },
  EMABERTO: {
  nomeEnum: "EM ABERTO",
  descricao: "Em Aberto",
  }
};
